test/helpers/runSolidityTest.js

```diff
@@ -55,34 +39,33 @@ const linkLib = async (contract, libName) => {
  * Runs a solidity test file, via javascript.
  * Required to smooth over some technical problems in solidity-coverage
  *
- * @param {string} c Name of solidity test file
+ * @param {string} testContract Name of solidity test file
 */
-function runSolidityTest(c, mochaContext) {
-  const artifact = artifacts.require(c)
-  contract(c, accounts => {
+function runSolidityTest(testContract, mochaContext) {
+  const artifact = artifacts.require(testContract)
+
+  contract(testContract, () => {
     let deployed
 
     before(async () => {
       await linkLib(artifact, 'Assert')
-
       deployed = await artifact.new()
     })
 
+    const assertResult = async (call, mustAssert) => {
+      const receipt = await call()
+      const { args: { result, message } } = getEventAt(receipt, 'TestEvent', { decodeForAbi: ASSERT_LIB_EVENTS_ABI })
+      if (!result) throw new Error(message || 'No assertions made')
+    }
+
     mochaContext('> Solidity test', () => {
       artifact.abi.forEach(interface => {
         if (interface.type === 'function') {
+          // Set up hooks
           if (['beforeAll', 'beforeEach', 'afterEach', 'afterAll'].includes(interface.name)) {
-            // Set up hooks
-            global[HOOKS_MAP[interface.name]](async () => {
-              const receipt = await deployed[interface.name]()
-              processResult(receipt, false)
-            })
+            global[HOOKS_MAP[interface.name]](async () => await deployed[interface.name]())
```

Comment:
Hmm, double checking here, does this fail properly if one of the hooks throws?

Reply:
In case some of these reverts, it throws. Note that previously `processResult` was called with `mustAssert` set to false always, therefore no assertions were actually being made, we were simply executing calls